PurchasesTests/DeviceCacheTests.swift

```diff
@@ -106,19 +63,35 @@ class DeviceCacheTests: XCTestCase {
     }
 
     func testResetCachesTimestamp() {
-        expect(self.deviceCache.isCacheStale()).to(beTrue())
-        self.deviceCache.resetCachesTimestamp()
-        expect(self.deviceCache.isCacheStale()).to(beFalse())
+        expect(self.deviceCache.isPurchaserInfoCacheStale()).to(beTrue())
+        self.deviceCache.setPurchaserInfoCacheTimestampToNow()
+        expect(self.deviceCache.isPurchaserInfoCacheStale()).to(beFalse())
     }
 
     func testCacheIsStaleIfNoCaches() {
-        expect(self.deviceCache.isCacheStale()).to(beTrue())
+        expect(self.deviceCache.isPurchaserInfoCacheStale()).to(beTrue())
+    }
+
+    func testPurchaserInfoCacheIsStaleIfLongerThanFiveMinutes() {
+        let oldDate: Date! = Calendar.current.date(byAdding: .minute, value: -(6), to: Date())
+        self.deviceCache = RCDeviceCache(mockUserDefaults, stubbedNow: oldDate)
+        self.deviceCache.cachePurchaserInfo(Data(), forAppUserID: "waldo")
+
+        expect(self.deviceCache.isPurchaserInfoCacheStale()).to(beFalse())
+        self.deviceCache.stubbedNow = Date()
```

Comment:
I might be missing something here, but Instead of setting this `stubbedNow`, can't you call `updateCacheTimestampWithDate` on the purchaser info cache and set it to a very old date?

Reply:
there isn't currently a public way to set the purchaser info's cache timestamp. 
it might not be a bad approach to make that public, but I wanted to restrict public access to everything as much as possible. 
the `stubbedNow` thing won't be accessible to customers unless they use `@testable import`, which they obviously shouldn't. 
maybe a good approach is to make certain properties accessible in the protected extension, that way `stubbedNow` isn't really needed. Thoughts?

Reply:
Sorry with public I meant not private to RCDeviceCache. The class
RCDeviceCahe is not accessible by customers either way.

What I was thinking about is exposing the purchaser info in the protected
extension. That way we don’t need the stubbednow. What’s the difference
between the testable import and the protected extension?